apps/gimbtonomy/LEDMatrix.h

```diff
@@ -0,0 +1,47 @@
+/* LED Matrix Header */
+
+#ifndef LED_MATRIX_H
+#define LED_MATRIX_H
+
+#include <mbed.h>
+#include <CANMsg.h>
+
+//Use whatever uint8_t red, green, blue gets assigned by the CAN messages to set the PWM of R, G, B 
+
+/*typedef union {
+	uint32_t  hexColourCode;
+		struct {
+			uint8_t green;
+			uint8_t blue;
+			uint8_t red;
+		} coloursRGB;
+} HexColours;*/
+
+class LEDMatrix {
+	
+	public:
+		//Data constructor - pins determined beforehand
+		//LEDMatrix();
+		
+		void setColour(uint8_t msgRed, uint8_t msgGreen, uint8_t msgBlue);
+
+		void setFrequency(uint8_t msgFrequency);
```

Comment:
I dont think it makes sense to expose setColour and setColourFrequency to the user of this lib. I might be wrong, but I dont see a case where the user would call these as oppose to setColourFrequency. It doesnt make sense to setFrequency but not color and vice versa

Reply:
Also, I believe whatever you expose in the api should return some kind of bool/enum indicating success.

Reply:
^ yes though this only makes sense if the functions called return something that indicates their failure or success or there is some logic that is evaluated that indicates that